storage: fix incorrect check in VersionContainsEstimatesCounter

tbg opened the pull request:
In RecomputeStats, we were supposed to avoid negative deltas until
they're allowed, but the code to do so was accidentally a noop.
This triggered a sanity check about the very same thing, which
in turn triggered a crash.

No release note since this was just introduced and won't be in anything
we release.

This failed in any test that consistency checked the timeseries range, I
used

make stress PKG=./pkg/server/ STRESSFLAGS='-p 4' TESTS=TestClusterVersionUpgrade

to convince myself, which would previously reproduce immediately.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/42273)
<!-- Reviewable:end -->

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42273#-:-Lt5efpOEJFEhR22FUvB:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @giorgosp and @tbg)

___
*[pkg/storage/replica_proposal.go, line 769 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/42273#-Lt5eVIS0hOdBzurEHPj:-Lt5eVIT1htk2vO1zjce:b-yhvdew) ([raw file](https://github.com/cockroachdb/cockroach/blob/9882f0bb19946ab36f0b5aa330508484c4c1def4/pkg/storage/replica_proposal.go#L769)):*
> ```Go
> 				pErr := roachpb.NewErrorf("cannot propose negative ContainsEstimates " +
> 					"without VersionContainsEstimatesCounter")
> 				log.Fatal(ctx, pErr)
> ```

Any reason not to just inline the fatal string? `log.Fatal(ctx, "cannot ...")`


<!-- Sent from Reviewable.io -->

giorgosp reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42273#-:-Lt5vjMkCF4Uaz0Xaxt1:bnfp4nl)* status: :shipit: complete! 2 of 0 LGTMs obtained (waiting on @giorgosp and @tbg)



<!-- Sent from Reviewable.io -->

tbg committed:
storage: fix incorrect check in VersionContainsEstimatesCounter

In RecomputeStats, we were supposed to avoid negative deltas until
they're allowed, but the code to do so was accidentally a noop.
This triggered a sanity check about the very same thing, which
in turn triggered a crash.

No release note since this was just introduced and won't be in anything
we release.

This failed in any test that consistency checked the timeseries range, I
used

make stress PKG=./pkg/server/ STRESSFLAGS='-p 4' TESTS=TestClusterVersionUpgrade

to convince myself, which would previously reproduce immediately.

Release note: None

tbg commented:
> Any reason not to just inline the fatal string? log.Fatal(ctx, "cannot ...")

Done.

TFTR!

bors r=nvanbenschoten,giorgosp

tbg commented:
Something busted in CI, going to try again later

```
[09:10:16]
Step 4/4: roachtest (Command Line)
[09:10:17]
[Step 4/4] Starting: /home/agent/temp/agentTmp/custom_script8998466241391208096
[09:10:17]
[Step 4/4] in directory: /home/agent/work/.go/src/github.com/cockroachdb/cockroach
[09:10:17]
[Step 4/4] Prepare environment
[09:10:17]
[Prepare environment]   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[09:10:17]
[Prepare environment]                                  Dload  Upload   Total   Spent    Left  Speed
[09:10:17]
[Prepare environment] 
[09:10:17]
[Prepare environment]   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[09:10:17]
[Prepare environment]   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[09:10:17]
[Prepare environment] curl: (22) The requested URL returned error: 503 Service Unavailable: Back-end server is at capacity
[09:10:17]
[Prepare environment] Process exited with code 22
[09:10:17]
[Step 4/4] Process exited with code 22 (Step: roachtest (Command Line))
[09:10:17]
[Step 4/4] Step roachtest (Command Line) failed
```

tbg commented:
bors r=nvanbenschoten,giorgosp
